app/controllers/PaymentHistoryController.scala

```diff
@@ -125,32 +116,15 @@ class PaymentHistoryController @Inject()(val paymentHistoryView: PaymentHistory,
       )
   }
 
-  private def handleStatusRefundRequest(isAgent: Boolean, itvcErrorHandler: ShowInternalServerError, backUrl: String)
-                                       (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
-    repaymentService.view(user.nino).flatMap {
-      case _ if !isEnabled(PaymentHistoryRefunds) =>
-        Future.successful(Ok(customNotFoundErrorView()(user, messages)))
-      case Right(nextUrl) =>
-        Future.successful(Redirect(nextUrl))
-      case Left(_) =>
-        Future.successful(itvcErrorHandler.showInternalServerError())
-    }
-
-  }
-
-  def refundStatus(isAgent: Boolean): Action[AnyContent] =
-    auth.authenticatedAction(isAgent) {
-      implicit user =>
-        user.userType match {
-          case _ if !isEnabled(PaymentHistoryRefunds) =>
-            Future.successful(Ok(customNotFoundErrorView()(user, user.messages)))
-          case Some(Agent) => Future.successful(itvcErrorHandlerAgent.showInternalServerError())
-          case _ =>
-            handleStatusRefundRequest(
-              backUrl = "",
-              itvcErrorHandler = itvcErrorHandler,
-              isAgent = false
-            )
+  def refundStatus: Action[AnyContent] =
+    authActions.asMTDIndividual.async { implicit user =>
+      if (isEnabled(PaymentHistoryRefunds)) {
+        repaymentService.view(user.nino).map {
+          case Right(nextUrl) => Redirect(nextUrl)
+          case Left(_) => itvcErrorHandler.showInternalServerError()
```

Comment:
I understand that this is not your code, but if we can add error logging as well